### ISSUE TITLE
Multiple names

### ISSUE THREAD

**Issue by gorstj** [bwssytems/ha-bridge#481]
Is it possible to have multiple names for the same 'light' (without making multiple copies)

E.g. Dining light could also be dining room light

I know I can do this in the Alexa app, but it would be more efficient to do it in HA-Bridge if I had more than one alexa

**Comment by bwssytems** [comment]
Not at this time. Maybe a future enhancement.

**Comment by smarty125** [comment]
Similar to #423.

**Comment by bwssytems** [comment]
Not really, #423 is for adding a nother field to describe the device that would not be presented to the api/echo, this is for aliases instead of creating another device doing the same command for a different name... i.e: TV, Television -> turn on TV

**Comment by bwssytems** [comment]
So, looking at this again, we would have to create a new Id and Unique Id to be stored with each name. Internally we will still be copying most of the data structure.

**Comment by bwssytems** [comment]
Ahh, just thought of a simpler way. It is kind of like a copy but is a special device that points to another device. That way maintenance of the details are only done once. But, you will still have a device entry as we need to have the new ID and Unique Id for the calling systems.

**Comment by Snorzze** [comment]
Any updates on this one? Would be a nice enhancement.

**Comment by AlfAlex** [comment]
I also would prefer something like this.
E.g. a new field (just a text area box) to enter alternative names for each device, one new name per row.
So I have several devices shown in Alexa but only one to be configured in HABridge.
This can be useful for devices like thermostats (also names as heating) or roller shutter or just shutter or roller and so on...

**Comment by bwssytems** [comment]
I think this can be done with a Groups or Alias as described in #1061

**Comment by AlfAlex** [comment]
I could be done like this maybe (but would be also a little bit effort).
But besides this this feature does not exist yet?
I think having the ability to add several names for one item and generate multiple devices  virtually later on would be much easier (to be developed).

**Comment by bwssytems** [comment]
@AlfAlex Unfortunately for the API to work properly to the Echo, we need to assign a unique ID to each name as that is how the an external system calls the Hue API to access a given light/device. i.e:

light1 has id 100
light2 has id 101
alias1 to light1 has an id of 102
alias2 to light2 has an id of 103

We cannot duplicate the id number to different names as the source system would be confused with this:

light1 has id 100
light2 has id 101
alias1 to light1 has an id of light1 of 100
alias2 to light2 has an id of light2 of 101

and the source system (Echo) would only show one name for the id 100 or 101.

**Comment by AlfAlex** [comment]
I understand.
But does it have to be an alias.
Just duplicate all entries would fit?
Wouldn't that work?

Let's say I have a light in my living room.
I want to turn it on by saying "turn on lamp in living room", "turn on light in living room", "turn on bulb in living room" and "turn on lamp in my blue room".
And all 4 commands will turn on the same light. So  there do exists four unique IDs for the same HTTP command.
I would imagine having the traditional name field and an additional alternate names field:

Name: ...
Alternate names: ...
 ...
...

HTTP command on:
HTTP command off:
HTTP command diff:
And so on.


Or isn't that possible?

**Comment by bwssytems** [comment]
It is just messy. If I create a device that points to one or more others that has it's own unique id it will be much cleaner for the code to respond to the request. Also, the Echo and other systems usually access the API to get a list of devices with their attributes to populate their control list.
